client/py/anvil/util.py

```diff
@@ -84,15 +84,19 @@ def __deepcopy__(self, memo):
 @serializable_type
 class WrappedList(list):
     def __init__(self, lst=[]):
-        for x in lst:
-            self.append(x)
+        if isinstance(lst, (WrappedObject, WrappedList)):
+            list.__init__(self, lst)
+        else:
+            list.__init__(self, map(_wrap, lst))
 
     def append(self, item):
         list.append(self, _wrap(item))
 
     def extend(self, items):
-        for i in items:
-            self.append(i)
+        if isinstance(items, (WrappedObject, WrappedList)):
```

Comment:
The short-circuit can also be simplified by using mixins. A bit easier to maintain, but sometimes people don't like multiple inheritance / diamonds... very much a case of taste